src/GraphQL/Execution/Nodes/ExecutionNode.cs

```diff
@@ -62,36 +62,16 @@ internal IGraphType ResolvedType
         /// </summary>
         public string Name => Field?.Alias ?? Field?.Name;
 
-        /// <summary>
-        /// Returns true if the result has been set. Also returns true when the result is temporarily set to an <see cref="IDataLoaderResult"/>
-        /// pending execution at a later time.
-        /// </summary>
-        public bool IsResultSet { get; private set; }
-
-        private object _result;
         /// <summary>
         /// Sets or returns the result of the execution node. May return a <see cref="IDataLoaderResult"/> if a node returns a data loader
         /// result that has not yet finished executing.
         /// </summary>
-        public object Result
-        {
-            get => _result;
-            set
-            {
-                IsResultSet = true;
-                _result = value;
-            }
-        }
+        public object Result { get; set; }
 
-        private object _source;
         /// <summary>
         /// Returns the parent node's result. If set, the set value will override the parent node's result.
         /// </summary>
-        public object Source
-        {
-            get => _source ?? Parent?.Result;
-            set => _source = value;
-        }
+        public virtual object Source => Parent?.Result;
```

Comment:
Only subscription executions change the source value.  Note that field resolvers do not have access to the setter, since `IResolveFieldContext.Source` is a read-only property.